src/NHibernate/Id/Insert/AbstractSelectingDelegate.cs

```diff
@@ -26,57 +26,67 @@ protected internal AbstractSelectingDelegate(IPostInsertIdentityPersister persis
 
 		public abstract IdentifierGeneratingInsert PrepareIdentifierGeneratingInsert();
 
-		public object PerformInsert(SqlCommandInfo insertSQL, ISessionImplementor session, IBinder binder)
+		public object PerformInsert(SqlCommandInfo insertSql, ISessionImplementor session, IBinder binder)
 		{
+			session.ConnectionManager.FlushBeginning();
```

Comment:
Significant changes are the addition of lines 31 to 33 and 87 to 89. Other changes are indentation or renaming.